### ISSUE TITLE
[Bug]: Only very short refund reason accepted in WooCommerce because of character maximum

### ISSUE THREAD

**Issue by JCZwier** [btcpayserver/btcpayserver#5398]
### What is your BTCPay version?

I'm running BTCPay v1.11.6+44a7e9387 (Altcoins).

### How did you deploy BTCPay Server?

I'm running BTCPay on Docker.

### What happened?

Refunding an order paid using BTCPay through WooCommerce is only possible when there is no reason for the refund entered, or when it is a very short string of characters (such as one word).  When a longer string is entered, an error comes up saying it should be less than 50 characters. This error happens already when entering way less characters (e.g. 20).

### How did you encounter this bug?

1. I initiated a refund through WooCommerce.
2. I entered a reason for the refund, consisting of a few words.
3. I clicked 'Refund via BTCPay' and  an error message appeared.

### Relevant log output

```shell
Exception creating pull payment: Error during POST to https://example.com/api/v1/stores/id/pull-payments. Got response (422): [{"path":"Name","message":"The name should be maximum 50 characters."}]
```

### What browser do you use?

Firefox 116.0

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

**Comment by ndeet** [comment]
Hi @JCZwier thanks for the report. You are right there is a limit of 50 chars on the refund name field which causes the error. The reason why there are not 50 chars is available is because we already add some text with the order id in there and add the input after that.

The default text was very long:
`Refund for order no.: ORDER_NO reason: REASON_INPUT`

I now cut it down to:
`Refund order ORDER_NO; REASON_INPUT`

So you will have more chars available and if you exceed them there will be no error but they will just get cut off for now. This is fixed with version 2.3.1 of the WooCommerce plugin.

The pull-request API added a separate description field a while ago, which we can use to put the reason but our PHP library does not support it yet and we can only enable that when we release v3 of that library. As soon as that happened I can update the WooCommerce plugin to use that field which is has at least 255 chars for the reason.

**Comment by ndeet** [comment]
@pavlenex this issue can be closed.
